flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/sinks/BatchSelectTableSink.java

```diff
@@ -49,19 +44,24 @@
  * once FLINK-14807 is finished, the implementation should be changed.
  */
 public class BatchSelectTableSink implements StreamTableSink<Row>, SelectTableSink {
+
 	private final TableSchema tableSchema;
-	private final String accumulatorName;
-	private final TypeSerializer<Row> typeSerializer;
-	private JobClient jobClient;
+	private final CollectSinkOperatorFactory<Row> factory;
+	private final CollectResultIterator<Row> iterator;
 
 	@SuppressWarnings("unchecked")
 	public BatchSelectTableSink(TableSchema tableSchema) {
```

Comment:
we can extract the most code of `BatchSelectTableSink` and `StreamSelectTableSink ` into a base class (SelectTableSinkBase)